src/daft-plan/src/optimization/rules/push_down_filter.rs

```diff
@@ -116,26 +115,55 @@ impl OptimizerRule for PushDownFilter {
                             return Ok(Transformed::No(plan));
                         }
                         let new_predicate = external_info.pushdowns().filters.as_ref().map(|f| predicate.and(f)).unwrap_or(predicate.clone());
-                        let partition_filter = if let ExternalInfo::Scan(ScanExternalInfo {scan_op,  ..}) = &external_info {
+                        let PredicateGroups { partition_filter, non_identity_partition_filter_for_data, partition_and_data_filter, data_only_filter } = if let ExternalInfo::Scan(ScanExternalInfo {scan_op,  ..}) = &external_info {
+                            // We split the predicate into four groups:
+                            // 1. Identity transform partition filters, which can then be dropped from the data-level filter.
```

Comment:
This subtle change here is that before we would only translate expressions to partition filters for allowlisted operations (<, <=, ==, >=, >, is_null, etc) but now it will allow all predicate expressions through. I think there may be some implications for this so we should think this through